### PR TITLE
Readme update and Terraform improvements

### PR DESCRIPTION
**Pull request by slilichenko**
# Description

Please provide a description of the changes you have made in this PR.

# How has this been tested?

Please explain how you have tested the new changes.

# Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have successfully run the E2E tests, and have included the links to the pipeline runs below
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have updated any relevant documentation to reflect my changes
- [ ] I have assigned a reviewer and messaged them

# Pipeline run links:

### CODE REVIEW

File: scripts/generate-tf-backend.sh [hunk 1]
```diff
@@ -58,36 +57,6 @@ section_open "Creating the service account for Terraform: tf-service-account"
...
-section_close
-
-section_open "Granting the service account permission to manage Cloud Storage"
-    gcloud projects add-iam-policy-binding "${TF_STATE_PROJECT}" \
-        --member serviceAccount:"${TF_SERVICE_ACCOUNT_NAME}"@"${TF_STATE_PROJECT}".iam.gserviceaccount.com \
-        --role roles/storage.admin
-section_close
-
-section_open "Enable the Cloud Resource Manager API with"
-    gcloud services enable cloudresourcemanager.googleapis.com
```
Comment (chmstimoteo) [inline scripts/generate-tf-backend.sh:74]: TF depends on this.

Comment (slilichenko) [reply scripts/generate-tf-backend.sh:74]: Fixed

File: infrastructure/README.md [hunk 2]
```diff
@@ -33,14 +33,25 @@ access control is desired multiple projects can be used:
...
+identity which is used in the Terraform scripts (your email or the dedicated service account email) the "Terraform
+principal" for brevity.
+
+The Terraform principal will need to be granted certain permissions in different projects:
+
+* the Owner role in all projects where the solution is to be installed. Required to install products related to the
+  solution.
+* the BigQuery [Data Owner role](https://cloud.google.com/bigquery/docs/control-access-to-resources-iam#required_roles)
+  on the datasets containing the GA4 and Ads data exports. Required to grant data read access to
+  a service account which w created by the Terraform scripts. Follow the
```
[the remark this answers is not in the document]
Comment (slilichenko) [reply infrastructure/README.md]: Fixed

File: scripts/generate-tf-backend.sh [hunk 3]
```diff
@@ -58,36 +57,6 @@ section_open "Creating the service account for Terraform: tf-service-account"
...
-    else
-        gsutil mb -p "${TF_STATE_PROJECT}" --pap enforced -l "${LOCATION}" -b on gs://"${TF_STATE_BUCKET}"
-        gsutil versioning set on gs://"${TF_STATE_BUCKET}"
-    fi
-section_close
-
-section_open "Creating terraform backend.tf configuration file"
-    TERRAFORM_RUN_DIR="infrastructure/terraform"
-    create_terraform_backend_config_file "${TERRAFORM_RUN_DIR}" "${TF_STATE_BUCKET}"
-section_close
```
[the remark this answers is not in the document]
Comment (slilichenko) [reply scripts/generate-tf-backend.sh]: Fixed

File: infrastructure/terraform/terraform-sample.tfvars [hunk 4]
```diff
@@ -27,25 +25,25 @@ deploy_pipelines     = true
...
 
 ####################  GITHUB VARIABLES  #################################
 
 project_owner_email   = "Project owner email"
-dataform_github_repo  = "URL of the GitHub or GitLab repo which contains the Dataform scripts"
+dataform_github_repo  = "URL of the GitHub or GitLab repo which contains the Dataform scripts. Should start with https://"
 dataform_github_token = "GitHub token generated for that repo"
-
-pipelines_github_owner = "Cloud Build github owner account for pipelines"
-pipelines_github_repo  = "Cloud Build github repository for pipelines"
```
Comment (kingman) [inline infrastructure/terraform/terraform-sample.tfvars:51]: pipelines_github_owner and pipelines_github_repo are use, can not be deleted

Comment (slilichenko) [reply infrastructure/terraform/terraform-sample.tfvars:51]: I provided fake default values in variables.tf

**Review by kingman — APPROVED** [review]

**Pull request merged**